networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationGossipManager.java

```diff
@@ -28,22 +27,21 @@
 
   private final GossipEncoding gossipEncoding;
   private final AttestationSubnetSubscriptions subnetSubscriptions;
-  private final EventBus eventBus;
 
   private final AtomicBoolean shutdown = new AtomicBoolean(false);
 
   public AttestationGossipManager(
       final GossipEncoding gossipEncoding,
-      final AttestationSubnetSubscriptions attestationSubnetSubscriptions,
-      final EventBus eventBus) {
+      final AttestationSubnetSubscriptions attestationSubnetSubscriptions) {
     this.gossipEncoding = gossipEncoding;
     subnetSubscriptions = attestationSubnetSubscriptions;
-    this.eventBus = eventBus;
-    eventBus.register(this);
   }
 
-  @Subscribe
-  public void onNewAttestation(final Attestation attestation) {
+  public void onNewAttestation(final ValidateableAttestation validateableAttestation) {
+    if (validateableAttestation.isGossiped() || validateableAttestation.isAggregate()) {
```

Comment:
Given you cleverly made `markGossiped` return a boolean, I'd do 
```suggestion
    if (validateableAttestation.isAggregate() || !validateableAttestation.markedGossiped()) {
```

Then you have no risk of two thread gossiping it at the same time.  In any case you need to markGossiped here to be sure we don't gossip it again later (we shouldn't with the current flow of things but that may change).

Reply:
Oh good catch! Done.

Reply:
Considering attestations will be one of our bigger loads, (especially already gossiped attestations) wouldn't quickly doing the `isGossiped()` check and then marking the attestation gossiped make sense?